lib/cell.rb

```diff
@@ -0,0 +1,42 @@
+class Cell
+  attr_reader :coordinate, :ship
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @hit_status = false
+  end
+
+  def empty?
+    @ship === nil
+  end
+
+  def place_ship(cruiser)
+    @ship = cruiser
+  end
+
+  def fire_upon
+    @hit_status = true
+    @ship.hit
+  end
+
+  def fired_upon?
+    @hit_status
+  end
+
+  def render(reveal_ship=false)
```

Comment:
`#Not sure if you are hitting all possibilities with the implementation, I would try this:
def render(reveal_ship=true)
  if @cell.fired_upon?
    if @cell.ship != nil && @cell.ship.health > 0
      "H"
    elsif @cell.ship != nil
      "X"
    else
      "M"
    end
  elsif reveal_ship && @cell.ship != nil
    "S"
  else
    "."
  end
end`

Reply:
Can you slack me this code snippet? GitHub's formatting is awful 😁